### PR TITLE
fix: 修复 Tag onClick 为 undefined，鼠标点击也会出现边框样式

### PR DESCRIPTION
**Pull request by crazyair**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Fix Tag onClick as undefined, click the mouse to display the border style |
| 🇨🇳 Chinese | 修复 Tag onClick 为 undefined，鼠标点击也会出现边框样式    |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: components/tag/index.tsx [hunk 1]
```diff
@@ -119,7 +119,7 @@ const InternalTag: React.ForwardRefRenderFunction<HTMLSpanElement, TagProps> = (
   };
 
   const isNeedWave =
-    'onClick' in props || (children && (children as React.ReactElement<any>).type === 'a');
+    props.onClick || (children && (children as React.ReactElement<any>).type === 'a');
```
Comment (yoyo837) [inline components/tag/index.tsx:122]: 个人感觉是不是`typeof`好些?

Comment (crazyair) [reply components/tag/index.tsx:122]: typescript 有约束 onClick 必须是 function，这里其实不怕是  true 字符串 之类的

Comment (yoyo837) [reply components/tag/index.tsx:122]: 但是antd也可以用js开发的呀

Comment (crazyair) [reply components/tag/index.tsx]: 行，严谨点吧

**Review by yoyo837 — APPROVED** [review]

File: components/tag/index.tsx [hunk 2]
```diff
@@ -119,7 +119,8 @@ const InternalTag: React.ForwardRefRenderFunction<HTMLSpanElement, TagProps> = (
   };
 
   const isNeedWave =
-    'onClick' in props || (children && (children as React.ReactElement<any>).type === 'a');
+    typeof props.onClick === 'function' ||
+    (children && (children as React.ReactElement<any>).type === 'a');
```
Comment (afc163) [inline components/tag/index.tsx:123]: 解决什么问题？

Comment (yoyo837) [reply components/tag/index.tsx:123]: @crazyair 来改一下 Changelog ?

Comment (crazyair) [reply components/tag/index.tsx:123]:
```tsx
<Tooltip>
    <Tag>dmeo</Tag>
</Tooltip>
```
hover 的 Tooltip，会给 children 注入 onClick，导致鼠标点击 Tag 会出现边框样式

**Pull request merged**